### PR TITLE
Bugfix/Fix update button being active without changes made to editable fields (edit member)

### PR DESCRIPTION
**Pull request by rfmajor**
Steps to reproduce:

Login as "8500032", "TestAccount1#"
Click Sell
Search customer by "testk2@example.com"
Click loyalty button
Click "Edit Member"
Don't make any changes
Expected result: 'Update' button should not be active.

Actual result: 'Update' button is active while no changes were made. 

**RCA**: Issue is happening because formGroup.pristine is always false whenever the dialog window appears
**Fix**: Add a boolean field to LoyaltyCustomerFormDialogComponent which is updated whenever the first call of onFieldChanged() happens and use it instead of formGroup.pristine

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/loyalty-customer-form-dialog/loyalty-customer-form-dialog.component.html [hunk 1]
```diff
@@ -156,7 +156,7 @@
                 <app-secondary-button *ngIf="screen.secondaryButton" (buttonClick)="secondaryButtonClicked()">
                     <span>{{screen.secondaryButton.title}}</span>
                 </app-secondary-button>
-                <app-primary-button *ngIf="screen.submitButton" type="submit" [disabled]="!screen.formGroup.valid || screen.formGroup.pending || (screen.formGroup.pristine && screen.enableSubmitButtonOnFormChange)">
+                <app-primary-button *ngIf="screen.submitButton" type="submit" [disabled]="!screen.formGroup.valid || screen.formGroup.pending || (isFormPristine && screen.enableSubmitButtonOnFormChange)">
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/loyalty-customer-form-dialog/loyalty-customer-form-dialog.component.html:159]: What was wrong with screen.formGroup.pristine? It should flip to false if any field in the group is changed

Comment (rfmajor) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/loyalty-customer-form-dialog/loyalty-customer-form-dialog.component.html:159]: It repetitively set to false without any prior changes from the user side. Not even formGroup.markAsPristine() can change that

Comment (rfmajor) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/loyalty-customer-form-dialog/loyalty-customer-form-dialog.component.html:159]: Here's the screenshot of the debugger right after clicking 'Edit member'. You can see on the right that pristine has a correct value at the end of ngOnInit()
image

And here's the screenshot after skipping to the next breakpoint (I've put breakpoints inside of all methods that belong to the class to see which one is being called next - hasLoyaltyOrMembershipsFields())
image

At this point I also went through the callstack to see if there are any calls that could potentially set this field to false, but I haven't found anything.

**Review by jchirico8 — APPROVED** [review]

**Comment by rfmajor** [comment]
@jchirico8 Sorry, I missed the jenkins check failure. Can you please approve once again? I needed to remove type annotation to satisfy lint requirements

**Review by jchirico8 — APPROVED** [review]

**Pull request merged**